### ISSUE TITLE
Consider removing react-virtualized for CSS Grid and position:sticky

### ISSUE THREAD

**Issue by techniq** [techniq/mui-table#7]
After seeing some examples by @souporserious using `display: grid` with `position:sticky`, they appear to track scroll position between fixed columns/rows better than react-virtualized's [MultiGrid](https://bvaughn.github.io/react-virtualized/#/components/MultiGrid).

- https://codesandbox.io/s/mjn5923qzj
- https://codesandbox.io/s/8xyvokv6p2
- https://codesandbox.io/s/xp4nk64qxq
- https://codesandbox.io/s/81rvm1qp22

IE 11 support for [display: grid](https://caniuse.com/#feat=css-grid) support is based on the old spec and likely not great.

IE 11 support for [position: sticky](https://caniuse.com/#search=sticky) is not available (might chalk it up to progressive enhancement).

Just considering for now.  Also watching the new [react-window](https://github.com/bvaughn/react-window)

**Comment by ericpyle** [comment]
I'm trying to use mui-table in context of an electron (Chrome) app. Would be nice to have a sticky header option (https://codesandbox.io/s/k0vwm7xpl3) since it's supported in Chrome. But virtualized grid may be sufficient for now.

**Comment by techniq** [comment]
@ericpyle I've been investigating a rewrite that drops react-virtualized and uses standard layout (`table`, `flexbox`, `grid`) in conjunction with `position: sticky`.  `position: sticky` appears to only allow a single `sticky` row/column at a time, but also supports "replacing" the stuck row/column which is nice.

I've been considering this for a while, as windowing does some limitations that I'm currently running into (need column spanning for example, which is non-trival with windowing) and mostly I paginate my tables and thus do not leverage windowing.

In short, I have some proof-of-concept working with table layout and position:sticky.  I'm likely to release this as `mui-table` `2.0.0` and release the current react-virtualized version as `mui-virtualized-table` or something similar as there is still some uses it handles better.  I'm planning to keep the props/apis as similar as possible (where it makes sense).

I'll let you know once I have an early beta available for testing if you're interested, and curious what you think of this split.

@cvanem You are likely to be interested this this as well.  No guarantees yet, but I have some use cases I need to solve soon, and this looks like the approach I plan to take.  I would like to enable using `react-window` with the new approach as much as I can (at least virtualize in the row direction, but maybe not columns as well)`.   This should also simplify simple cases of row hovering/selecting, etc, and make them more performance (but I still plan to support at this time the `isCellSelected` and `isCellHovered` props as they give good control, at the expense of some performance (but could be improved with memoization/etc).

**Comment by cvanem** [comment]
@techniq OK, I think either implementation works for me.  I can try out the beta, once it is ready, if you would like.

**Comment by techniq** [comment]
@ericpyle @cvanem 

- Released mui-table version `2.0.0` which uses `display: table`.  See https://github.com/techniq/mui-table/issues/13 for details on the release
- Created [mui-virtualized-table](https://github.com/techniq/mui-virtualized-table) (which was mui-table `1.x.x`) and released version `1.0.1`

I haven't had a chance to consume either in my projects yet (I will be tomorrow) and still plan to investigate more functionality (`display: grid`, etc).  Let me know how either works for you and any feedback you have.

**Comment by ericpyle** [comment]
@techniq looking good! Thanks so much! 

Btw. I made a PR for you to address one exception I encountered when trying to configure sortable column headers.

https://github.com/techniq/mui-table/pull/17
